thunder/executors/nvfuserex_impl.py

```diff
@@ -2093,7 +2093,11 @@ def var_mean(
 def _copy__check(
     copy_from: TensorProxy,
     copy_to: TensorProxy,
+    *,
+    grad_enabled: bool,
 ) -> bool:
+    if grad_enabled:
+        warnings.warn("An inplace operation requiring grad is occurring on a variable which may be a leaf.")
```

Comment:
We can easily query at the construction time of input TensorProxy whether the corresponding PyTorch tensors have `is_leaf` set to True similar to how we do this for `requires_grad`:
https://github.com/Lightning-AI/lightning-thunder/blob/eff35942f52156345629a04182564efe9e209782/thunder/core/proxies.py#L1993

With this information on the proxy, we can raise a real error.

@beverlylytle, could you please create an issue about this to discuss whether it's a good idea to do so?

Reply:
done. #1577 1577